frontend/src/tests/components/Courses/CoursesForm.test.js

```diff
@@ -0,0 +1,114 @@
+import { render, waitFor, fireEvent, screen } from "@testing-library/react";
+import CoursesForm from "main/components/Courses/CoursesForm";
+import { coursesFixtures } from "fixtures/coursesFixtures";
+import { BrowserRouter as Router } from "react-router-dom";
+
+const mockedNavigate = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+    ...jest.requireActual('react-router-dom'),
+    useNavigate: () => mockedNavigate
+}));
+
+
+describe("CoursesForm tests", () => {
+
+    test("renders correctly", async () => {
+
+        render(
+            <Router  >
+                <CoursesForm />
+            </Router>
+        );
+        await screen.findByText(/Name/);
+        await screen.findByText(/Create/);
+    });
+
+
+    test("renders correctly when passing in a Courses", async () => {
+
+        render(
+            <Router  >
+                <CoursesForm initialContents={coursesFixtures.oneCourse} />
+            </Router>
+        );
+        await screen.findByTestId(/CoursesForm-id/);
+        expect(screen.getByText(/Id/)).toBeInTheDocument();
+        expect(screen.getByTestId(/CoursesForm-id/)).toHaveValue("1");
+    });
+
+
+    test("Correct Error messsages on missing input", async () => {
```

Comment:
(nitpick) **messages**

Reply:
Ill probably fix this in another PR but good catch!